models/repo.go

```diff
@@ -2389,6 +2391,36 @@ func CheckRepoStats() {
 	// ***** END: Repository.NumForks *****
 }
 
+// ToggleArchiveRepo changes the status if a repo is archived or not
+func (repo *Repository) ToggleArchiveRepo() (err error) {
+	repo.IsArchived = !repo.IsArchived
+	_, err = x.Where("id = ?", repo.ID).Cols("is_archived").Update(repo)
+
+	// Enable/Disable issues and pull requests
+	if repo.IsArchived {
+		if _, err = x.
```

Comment:
They should be view-mode only, but it's unnecessary to delete the units.

Reply:
I know, moved it to a middleware.